src/spec-lang/tc/typecheck.ts

```diff
@@ -1297,17 +1322,27 @@ export function tcMemberAccess(expr: SMemberAccess, ctx: STypingCtx, typeEnv: Ty
     const baseT = tc(expr.base, ctx, typeEnv);
 
     if (baseT instanceof BuiltinStructType) {
-        const type = baseT.members.get(expr.member);
+        const result = baseT.members.get(expr.member);
 
-        if (type === undefined) {
+        if (result === undefined) {
             throw new SNoField(
                 `Builtin struct "${expr.base.pp()}" does not have a member "${expr.member}"`,
                 expr,
                 expr.member
             );
         }
 
-        return type;
+        if (result instanceof Array) {
```

Comment:
There is some code duplication between this and `tcIdBuiltinSymbol`. Can we abstract in a helper function?

Reply:
Addressed in https://github.com/ConsenSys/scribble/pull/45/commits/3fda5277eb180bda7d656fc0d036f878c21290fe